addons/pos_restaurant/static/src/js/Screens/TipScreen.js

```diff
@@ -67,26 +67,61 @@ odoo.define('pos_restaurant.TipScreen', function (require) {
                 });
                 if (!confirmed) return;
             }
-
+            /**
+             * Callback to be notified, when all the rewards are finish being updated.
+             * kept here to have access to the variables used within `validateTip`.
+             */
+             const _callbackAfterTip = async () => {
+                /**
+                 * We only want this to be notified once so we insta-clear this callback.
+                 * Also in this point we can safely set the order back to `finalzed = true`,
+                 * since all the rewards has been applied.
+                 */
+                order.off('rewards-updated', _callbackAfterTip);
```

Comment:
esta linea no deberia estar bajo un 
if (this.env.pos.config.use_coupon_programs) ?

Reply:
En si no hay ningun probema con intentar remover ese listener. No hay tracebacks por eventos "no definidos", ya hice la prueba con ambos casos y jalo al pelo